std/assembly/typedarray.ts

```diff
@@ -16,14 +24,22 @@ export class Uint8Array extends TypedArray<u8,u32> {
   subarray(begin: i32 = 0, end: i32 = 0x7fffffff): Uint8Array {
     return changetype<Uint8Array>(super.subarray(begin, end));
   }
+  map(callbackfn: (value: u8, index: i32, array: Uint8Array) => u8): Uint8Array {
+    var length = this.length;
+    var result = new Uint8Array(this.length);
+    for (let i = 0; i < length; ++i) {
+      result.__unchecked_set(i, callbackfn(this.__unchecked_get(i), i, this));
+    }
+    return result;
+  }
 }
 
 export class Uint8ClampedArray extends TypedArray<u8,u32> {
   static readonly BYTES_PER_ELEMENT: usize = sizeof<u8>();
 
   @inline @operator("[]=")
   protected __set(index: i32, value: i32): void {
-    super.__set(index, max(min(value, 255), 0));
+    super.__unchecked_set(index, max(min(value, 255), 0));
```

Comment:
changing this function back to `super.__set` causes this error to occur when testing `compiler/std/typedarray.ts`

```
Fatal: Module::addFunction: $~lib/internal/typedarray/TypedArray<u8,u32>#__set already exists
```

Reply:
Try renaming the function itself to `__setClamped` as a workaround.

Reply:
![image](https://user-images.githubusercontent.com/3761339/49187838-c98c5180-f336-11e8-9b15-459164abe071.png)

Reply:
I'm so confused. The test was perfectly fine before the change.

Reply:
may be you forget replace `result.__set` to `result.__setClamped` for `map` inside `Uint8ClampedArray`?

Reply:
As it turns out, I don't think that matters because the values will already be clamped, and changing the function names simply does not resolve the issue.

![image](https://user-images.githubusercontent.com/3761339/49188401-a2368400-f338-11e8-9b73-03571e080fe7.png)

![image](https://user-images.githubusercontent.com/3761339/49188414-b11d3680-f338-11e8-9faf-5f1d2760b471.png)

It is trying to add the `__set` method to the table when it already exists apparently. Could this be a compiler error?

Reply:
I can bring this chat to slack if you want, but I'll be around to work on this again tomorrow. Thank you @MaxGraey and @dcodeIO  for your amazing help.

Reply:
Yes, this is most likely a compiler error that involves `super`.

Reply:
Where can I look to help?